### ISSUE TITLE
Persisting the old enviornment and generating the old enviornment with the same html in the a-scene 

### ISSUE THREAD

**Issue by lazychip** [supermedium/aframe-environment-component#51]
When i try to generate the old environment by saving it and injecting it  in the a-scene environment is broken.

**Comment by feiss** [comment]
sorry, I don't understand what you are doing exacly. Could you explain?

**Comment by lazychip** [comment]
I want to save the scene state and reload with the saved scene state
Step 1 => flush the dom
Step 2 => save the scene in the local storage
Step 3 => on reload get the scene state from the local storage and load the scene.
So what happening is everything is working fine but the environment is not loading properly

**Comment by feiss** [comment]
how are you saving the scene?

**Comment by lazychip** [comment]
saving the entities and assets as the separate arrays
making a template from the saved entities and assets to load the scene back

what i did to make environment work
1. save the outerHTML only not the child entities(innerHTML)
2. this repo has default values for the specific preset so what i did is just save the difference of the default values and the flushed environment that gives me the only changed properties of the environment that has been made through the inspector.

I know this is a hack there should be more simple and consistent way of doing it.

**Comment by lazychip** [comment]
`
const t = entity.cloneNode(true);
const differenceInAttr = difference(
  entity.getAttribute('environment'),
  environmentDefaultProps[entity.getAttribute('environment').preset],
);
let attrInString = '';
Object.keys(differenceInAttr).forEach(attr => {
  attrInString += `${attr}:${differenceInAttr[attr]};`;
});
t.attributes.environment.value = attrInString;
t.innerHTML = '';
entitiesString.push(t.outerHTML);
`

**Comment by feiss** [comment]
Yeah i think all this is unnecesary complex. If you call to the `dumpParametersDiff()` function of the component:

```javascript
document.querySelector('[environment]').components['environment'].dumpParametersDiff();
```
you will get the settings you need to recreate the same environment:
```javascript
 document.querySelector('[environment]').setAttribute('environment', settings);
```
I just realized that `dumpParametersDiff()` only dumps to console, doesn't return an object with the settings so you can save them programatically. I can fix that if you feel this solves your problem.

**Comment by lazychip** [comment]
Yeah sure
